lib/task-graph.js

```diff
@@ -88,28 +132,22 @@ function taskGraphFactory(
         if (marker.permanentMark) {
             return;
         }
-
         marker.temporaryMark = true;
-
+        
         _.forEach(task.waitingOn, function(value, dep) {
             var nonTerminalOnStates = [];
-
-            // This won't actually get called because we do this same check in
-            // _populateTaskData, but keep this here just to be defensive.
-            if (!_.has(self.tasks, dep)) {
-                throw new Error('Graph does not contain task with ID ' + dep);
-            }
-            // Expand 'finished' to all finished task states
-            if (_.contains(value, Constants.Task.States.Finished)) {
-                value = Constants.Task.FinishedStates;
-                task.waitingOn[dep] = [Constants.Task.States.Finished];
+            if (dep === "either") {
```

Comment:
The word `either` is a magic value or a keyword in workflow definition, We can either defining some indicator to distinguish it from task labels, such as changing `either` to `$either`, or we need add a validation logic in workflow definition to make sure the `either` is not been used as task label.